Testing/Readme.md

```diff
@@ -0,0 +1,424 @@
+Export-Certificate -Cert $MyCert -Type CERT -FilePath .\ScubaServicePrincipal.cer
+```
+Once the certificate is created, provide it to the tenant administrator to align it with functional testing application. 
+
+## Functional Testing Structure ##
+
+ScubaGear functional testing suite has two main components: the orchestrator and product test plans.
+
+### Functional test orchestrator ### 
+
+Functional test orchestrator (`/Testing/Functional/Products/Products.tests.ps1`) executes the tests using a predefined product test plan and various input test parameters as defined by the tester. These parameters are:
+
+- Thumbprint of the certificate associated with functional testing service principal
+- AppId associated with service principal and certificate
+- Tenant Domain
+- Tenant Display Name
+- M365 Environment type (viz. `gcc`, `gcchigh`, `commercial`)
+- An optional variant parameter
+
+### Product test plans ###
+
+Each of the ScubaGear supported products should have one or more test plans. Each test plan is defined by a YAML file and consists of multiple test cases for each secure configuration policy. Test cases are defined for both non-compliant and compliant use cases. The objective of functional testing is to test every possible non-compliant/compliant configuration for a policy and ensure that ScubaGear is correctly identifying it as a policy failure/pass. 
+
+The development teams for each product are responsible for maintaining these test plans and ensuring that they are accurate and complete. Note that each product has at least one test plan associated with it. There can be multiple test plans for a product; these additional test plans are used to capture unique configurations based on tenant type and/or PowerShell variant. To differentiate among the test plans, they use the following naming convention:
+```
+product.<variant>.testplan.yaml
+```
+The optional variant can have values like: `g5`, `e5`, `gcc`, `pnp`, `spo`, `e#` etc. These `variant` test plans capture test cases unique to that tenant type and/or powershell type. 
+
+
+
+## Functional Testing Usage ##
+
+After setting up the prerequisites, define the Pester test container parameters a test execution utility script called "RunFunctionalTest.ps1" as shown below: 
+
+
+```
+$TestContainers = @()
+
+$TestContainers += New-PesterContainer -Path "Testing/Functional/Products" -Data @{ TenantDomain = "MyG5tenant.onmicrosoft.com"; TenantDisplayName = "My G5 tenant"; ProductName = "aad"; M365Environment = "gcc" }
+
+$PesterConfig = @{
+	Run = @{
+		Container = $TestContainers
+	}
+	Output = @{
+		Verbosity = 'Detailed'
+	}
+}
+
+$Config = New-PesterConfiguration -Hashtable $PesterConfig 
+
+Invoke-Pester -Configuration $Config
+```
+Copy the above utility script and save it into a file named "RunFunctionalTest.ps1" on your system. Save this file in a folder named FunctionalTesting that is at the same level in your directory tree as ScubaGear as shown below.
```

Comment:
It should be noted that wherever one stores the utility script, it's important to run it script from the root of the ScubaGear directory, not from the directory where the utility script is stored.  Perhaps this could be added as a note.

Reply:
added.